ddtrace/internal/telemetry/writer.py

```diff
@@ -216,26 +234,27 @@ def _create_telemetry_request(self, payload, payload_type, sequence_id):
             "request_type": payload_type,
         }
 
-    def _restart(self):
+    def _fork_writer(self):
         # type: () -> None
-        if self.status == ServiceStatus.RUNNING:
-            self.disable()
-        self.enable()
+        self._forked = True
+        # Avoid sending duplicate events.
+        # Queued events should be sent in the main process.
+        self._reset_queues()
+        self.start()
 
     def disable(self):
         # type: () -> None
         """
         Disable the telemetry collection service and drop the existing integrations and events
         Once disabled, telemetry collection can be re-enabled by calling ``enable`` again.
         """
-        with self._lock:
-            self._integrations_queue = []
-            self._enabled = False
-            if self.status == ServiceStatus.STOPPED:
-                return
+        self._reset_queues()
```

Comment:
We should lock here as well. All-to-easy to have a race with something here and in the thread (eg. on `self._enabled`)